proposals/dao/vip_2.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  DeployUpgradeFunc,
+  NamedAddresses,
+  SetupUpgradeFunc,
+  TeardownUpgradeFunc,
+  ValidateUpgradeFunc
+} from '@custom-types/types';
+import config from '../../scripts/config';
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { getImpersonatedSigner } from '@test/helpers';
+const { STARTING_ORACLE_PRICE, ORACLE_PERIOD_START_TIME, ANNUAL_CHANGE_RATE_BASIS_POINTS } = config;
+
+/*
+
+Volt Protocol Improvement Proposal #2
+
+Description: Patch in a new system oracle and repoint all PSM's to it
+
+Steps:
+  1 - Deploy Volt System Oacle
+  2 - Deploy Oracle Pass Through
+  3 - Grant ownership of Oracle Pass Through to the timelock
+  4 - Point both FEI and USDC PSM to the new OraclePassThrough
+
+*/
+
+const vipNumber = '2';
+
+const deploy: DeployUpgradeFunc = async (deployAddress: string, addresses: NamedAddresses, logging: boolean) => {
+  const { optimisticTimelock } = addresses;
+  const VoltSystemOracleFactory = await ethers.getContractFactory('VoltSystemOracle');
+  const OraclePassThroughFactory = await ethers.getContractFactory('OraclePassThrough');
+
+  const voltSystemOracle = await VoltSystemOracleFactory.deploy(
+    ANNUAL_CHANGE_RATE_BASIS_POINTS,
+    ORACLE_PERIOD_START_TIME,
+    STARTING_ORACLE_PRICE
+  );
+  await voltSystemOracle.deployed();
+
+  const oraclePassThrough = await OraclePassThroughFactory.deploy(voltSystemOracle.address);
+  await oraclePassThrough.deployed();
+
+  await oraclePassThrough.transferOwnership(optimisticTimelock);
+
+  console.log(`Deployed volt system oracle and oracle passthrough VIP-${vipNumber}`);
+  return {
+    voltSystemOracle,
+    oraclePassThrough
+  };
+};
+
+const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {};
+
+// Tears down any changes made in setup() that need to be
+// cleaned up before doing any validation checks.
+const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {};
+
+// Run any validations required on the vip using mocha or console logging
+// IE check balances, check state of contracts, etc.
+const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  const { core, oraclePassThrough, voltSystemOracle, feiPriceBoundPSM, usdcPriceBoundPSM, optimisticTimelock } =
+    contracts;
+
+  const { pcvGuardEOA1, pcvGuardEOA2, pcvGuardEOA3, pcvGuardRevoked1 } = addresses;
+
+  expect(await core.isGuardian(pcvGuardEOA1)).to.be.true;
+  expect(await core.isGuardian(pcvGuardEOA2)).to.be.true;
+  expect(await core.isGuardian(pcvGuardEOA3)).to.be.true;
```

Comment:
revoke pcv guardian role from revoked EOA

Reply:
move away from PCV Guardian role, possibly revoking it for now from all EOA's as all PCV Guardians are also Guardians

Reply:
This is a separate PR